core/app/modules/kit-library/assets/js/app.js

```diff
@@ -26,7 +25,7 @@ export default function App() {
 				<SettingsProvider value={ elementorAppConfig[ 'kit-library' ] }>
 					<Router>
 						<Index path="/" initialQueryParams={{}}/>
-						<Index path="/favorites" initialQueryParams={{ favorite: true }} noResultComponent={IndexFavoriteNoResult}/>
+						<Index path="/favorites" initialQueryParams={{ favorite: true }}/>
```

Comment:
Separate into 2 components. from the favorites return the index with all favorites-related props.